src/attached_probe.cpp

```diff
@@ -804,6 +804,22 @@ void AttachedProbe::load_prog(BPFfeature &feature)
         throw std::runtime_error(errmsg.str());
       }
     }
+
+    // try to identify an illegal helper call
+    if (std::string(log_buf.get())
+            .find("helper call is not allowed in probe") != std::string::npos)
+    {
+      // get helper function name and id
+      std::regex re("[0-9]+: \\([0-9]+\\) call ([a-z|A-Z|0-9|_]+)#([0-9]+)");
```

Comment:
Maybe pull this into a global in an anon namespace? So we don't reconstruct the regex every time this code path is hit. Making this static also works but IIRC the compiler has to generate some locks to synchronize construction between multiple threads